homeassistant/core.py

```diff
@@ -1092,100 +1087,60 @@ def call(self, domain: str, service: str,
-                EVENT_CALL_SERVICE, event_data, EventOrigin.local, context)
+            self._hass.async_create_task(
+                self._catched_execute(handler, service_call))
             return None
 
-        fut = asyncio.Future()  # type: asyncio.Future
-
-        @callback
-        def service_executed(event: Event) -> None:
-            """Handle an executed service."""
-            if event.data[ATTR_SERVICE_CALL_ID] == call_id:
-                fut.set_result(True)
-                unsub()
-
-        unsub = self._hass.bus.async_listen(
-            EVENT_SERVICE_EXECUTED, service_executed)
-
-        self._hass.bus.async_fire(EVENT_CALL_SERVICE, event_data,
-                                  EventOrigin.local, context)
-
-        done, _ = await asyncio.wait([fut], timeout=SERVICE_CALL_LIMIT)
-        success = bool(done)
-        if not success:
-            unsub()
-        return success
-
-    async def _event_to_service_call(self, event: Event) -> None:
-        """Handle the SERVICE_CALLED events from the EventBus."""
-        service_data = event.data.get(ATTR_SERVICE_DATA) or {}
-        domain = event.data.get(ATTR_DOMAIN).lower()  # type: ignore
-        service = event.data.get(ATTR_SERVICE).lower()  # type: ignore
-        call_id = event.data.get(ATTR_SERVICE_CALL_ID)
-
-        if not self.has_service(domain, service):
-            if event.origin == EventOrigin.local:
-                _LOGGER.warning("Unable to find service %s/%s",
-                                domain, service)
-            return
-
-        service_handler = self._services[domain][service]
-
-        def fire_service_executed() -> None:
-            """Fire service executed event."""
-            if not call_id:
-                return
-
-            data = {ATTR_SERVICE_CALL_ID: call_id}
-
-            if (service_handler.is_coroutinefunction or
-                    service_handler.is_callback):
-                self._hass.bus.async_fire(EVENT_SERVICE_EXECUTED, data,
-                                          EventOrigin.local, event.context)
-            else:
-                self._hass.bus.fire(EVENT_SERVICE_EXECUTED, data,
-                                    EventOrigin.local, event.context)
-
         try:
-            if service_handler.schema:
-                service_data = service_handler.schema(service_data)
-        except vol.Invalid as ex:
-            _LOGGER.error("Invalid service data for %s.%s: %s",
-                          domain, service, humanize_error(service_data, ex))
-            fire_service_executed()
-            return
-
-        service_call = ServiceCall(
-            domain, service, service_data, event.context)
+            with timeout(SERVICE_CALL_LIMIT):
+                await asyncio.shield(
+                    self._execute_service(handler, service_call))
+            return True
+        except asyncio.TimeoutError:
+            return False
 
+    async def _catched_execute(self, handler: Service,
```

Comment:
The name doesn't really roll of my tongue. :wink:
What about `_execute_safely`?

Reply:
I don't like either

Reply:
_safe_execute() ?

Reply:
deal